lib/find_retryables.ts

```diff
@@ -95,23 +107,43 @@ const main = async () => {
 
       const messages = await arbL2TxReceipt.getL1ToL2Messages(l3Provider)
 
+      if (messages.length == 0) {
+        console.log(`No retryable is found!`)
+        break
+      } else {
+        console.log(
+          `${messages.length} retryable${
+            messages.length === 1 ? '' : 's'
+          } found, checking their status. Arbtxhash: ${process.env
+            .ARBISCAN!}${l2TxHash}`
+        )
+        console.log('************************************************')
+      }
+
       for (let msgIndex = 0; msgIndex < messages.length; msgIndex++) {
```

Comment:
Still see duplication of retryables and I think it comes from here. In this "for" loop, the script iterates through all messages that are retryables, for the transaction of every inboxDeliveredLog. The script is picking 4 inboxDeliveredLogs coming from the same transaction, but it iterates over all of them, yielding the same 4 retryables 4 times.

One alternative to fix this is that In a first for loop (for inboxDeliveredLogs), you gather all l2TxReceipt in an array, and then, outside the for loop, you traverse that new array and perform the for loop for the "messages".